invenio_circulation/transitions/transitions.py

```diff
@@ -157,12 +157,19 @@ def before(self, loan, **kwargs):
         _ensure_valid_loan_duration(loan)
 
 
-class ItemAtDeskToItemOnLoan(ToItemOnLoan):
+class ItemAtDeskToItemOnLoan(Transition):
     """Check-out action to perform a loan when item ready at desk."""
 
     def before(self, loan, **kwargs):
         """Validate checkout action."""
         super().before(loan, **kwargs)
+
+        self.ensure_at_desk_item_is_available_for_checkout(loan)
+
+        if not kwargs.get("pickup_location_pid") \
```

Comment:
Minor: this is a small repetition of the code in the `before` method in `ToItemOnLoan`. Personally, I am ok with that, I am thinking now that you might want to put it in common by calling a method, e.g `set_default_pickup_location`.